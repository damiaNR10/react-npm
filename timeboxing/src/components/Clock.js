import React from 'react';

function Clock({className ="", hours = 1, minutes = 2, seconds = 3, miliseconds = 4}) {
    return <h2 className={"Clock " + className}>PozostaÅ‚o {minutes > 9 ? minutes : "0" + minutes}:{seconds > 9 ? seconds : "0" + seconds}</h2>;
}

Clock.defaultProps = {
    className: "",
    hours: 1,
    minutes: 2,
    seconds: 3,
    miliseconds: 4
};

export default Clock;