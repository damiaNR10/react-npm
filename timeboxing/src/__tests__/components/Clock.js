import React from 'react';
import Clock from '../../components/Clock';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

let root = null;
let clockRenderer = null;
describe("<Clock />", () => {

    describe("when given minutes and seconds (DOM)", () => {

        beforeEach(() =>{
            root = document.createElement("div");
            ReactDOM.render(
                <Clock minutes = {20} seconds = {40} />, root
            );
        });

        it("renders properly", () => {
            expect(root.childNodes[0].nodeName).toEqual("H2");
            expect(root.childNodes[0].className).toMatch(/Clock/);
            expect(root.childNodes[0].textContent).toMatch(/20:40/);
        });

        it("renders h2 element properly", () => {
            expect(root.childNodes[0].nodeName).toEqual("H2");
        });
    
        it("sets a Clock className", () => {
            expect(root.childNodes[0].className).toMatch(/Clock/);
        });
    
        it("renders time properly", () => {
            expect(root.childNodes[0].textContent).toMatch(/20:40/);
        });
    });

    describe("when given minutes and seconds (TestRenderer)", () => {

        beforeEach(() =>{
            clockRenderer = renderer.create(
                <Clock minutes = {20} seconds = {40} />
            )
        });

        it("renders properly", () => {
            expect(clockRenderer.toJSON()).toMatchSnapshot();
            // expect(clockRenderer.toJSON().type).toEqual('h2');
            // expect(clockRenderer.toJSON().props).toMatchObject({'className': expect.stringMatching(/Clock/)});
            // expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["20", "40"]));
        });

        it("renders h2 element properly", () => {
            expect(clockRenderer.toJSON().type).toEqual('h2');
        });
    
        it("sets a Clock className", () => {
            expect(clockRenderer.toJSON().props).toMatchObject({'className': expect.stringMatching(/Clock/)});
        });
    
        it("renders time properly", () => {
            expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["20", "40"]));
        });
    });

    it("sets className to empty string if not given anything else", () => {
        expect(<Clock minutes = {20} seconds = {40} />).toEqual(<Clock minutes = {20} seconds = {40} />);
    });
});