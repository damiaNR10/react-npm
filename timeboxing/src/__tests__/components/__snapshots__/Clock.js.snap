// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (TestRenderer) renders properly 1`] = `
<h2
  className="Clock "
>
  Pozosta≈Ço 
  20
  :
  40
</h2>
`;
